### PR TITLE
[cga] Removed support for cub::CachingDeviceAllocator

### LINKED ISSUE #399: [common] remove CUB caching allocator
Since CUB caching allocator is not being actively used in the SDK and there are semantic differences in how the CUB and preallocator allocator APIs behave, we decided to remove the CUB allocator for now. If need be, it can be brought back from git history.

For some insight into CUB semantics issue, please see https://github.com/clara-genomics/ClaraGenomicsAnalysis/pull/379

### PR DESCRIPTION
**Pull request by mimaric**
- Removed support for `cub::CachingDeviceAllocator` as discussed in PR #379 

- Removed `device_buffer.cuh`. This is an old implementation of `DeviceBuffer`. The whole codebase has been migrated to `deivce_buffer.hpp`

Closes #399

### CODE REVIEW

**Comment by tijyojwad** [comment]
Closes #399

File: common/base/CMakeLists.txt [hunk 1]
```diff
@@ -29,11 +29,7 @@ if (cga_device_synchronize_kernels)
 endif()
 
 if(cga_enable_caching_allocator)
-        if (cga_enable_preallocating_caching_allocator)
-            target_compile_definitions(${PROJECT_NAME} PUBLIC CGA_ENABLE_CACHING_ALLOCATOR CGA_ENABLE_PREALLOCATING_ALLOCATOR)
-        else()
             target_compile_definitions(${PROJECT_NAME} PUBLIC CGA_ENABLE_CACHING_ALLOCATOR)
```
Comment (tijyojwad) [inline common/base/CMakeLists.txt:32]: the indent here needs to be updated now that the inner conditional has been removed

Comment (mimaric) [reply common/base/CMakeLists.txt]: done

Comment (tijyojwad) [reply common/base/CMakeLists.txt]: Thanks!

**Pull request merged**